crates/nu-cli/src/repl.rs

```diff
@@ -420,7 +420,7 @@ fn loop_iteration(ctx: LoopContext) -> (bool, Stack, Reedline) {
     let style_computer = StyleComputer::from_config(engine_state, &stack_arc);
 
     start_time = std::time::Instant::now();
-    line_editor = if config.use_ansi_coloring.get(engine_state) {
+    line_editor = if config.use_ansi_coloring.get(engine_state) && config.show_hints {
```

Comment:
I guess whether hints are shown depends on more than just the value of `show_hints`; perhaps it should be a `disable_hints` option instead. Alternatively, the documentation could say something along the lines of: show hints (if supported).